internal/dag/builder.go

```diff
@@ -354,11 +354,15 @@ func (b *Builder) computeIngressRoutes() {
 			// attach secrets to TLS enabled vhosts
 			m := splitSecret(tls.SecretName, ir.Namespace)
 			sec := b.lookupSecret(m, validSecret)
-			if sec != nil && b.delegationPermitted(m, ir.Namespace) {
-				svhost := b.lookupSecureVirtualHost(host)
-				svhost.Secret = sec
-				svhost.MinProtoVersion = MinProtoVersion(ir.Spec.VirtualHost.TLS.MinimumProtocolVersion)
-				enforceTLS = true
+			if sec != nil {
+				if b.delegationPermitted(m, ir.Namespace) {
+					svhost := b.lookupSecureVirtualHost(host)
+					svhost.Secret = sec
+					svhost.MinProtoVersion = MinProtoVersion(ir.Spec.VirtualHost.TLS.MinimumProtocolVersion)
+					enforceTLS = true
+				} else {
+					b.setStatus(Status{Object: ir, Status: StatusInvalid, Description: fmt.Sprintf("%s: certificate delegation not permitted", tls.SecretName)})
```

Comment:
Do we need a `continue` after this, to stop processing the IngressRoute?

Reply:
technically yes, but the next condition, `sec == nil` will prevent the other 1/2 of this function running. Having said that, this is a horrid piece of nested logic. Let me see if I can clean it up some so we can use `continue` or `return` after a status setting.